src/pages/Dashboard/components/TodosByGoal/TodosByGoalBox.tsx

```diff
@@ -1,134 +1,114 @@
-            height="100%"
+        <div className="mt-2 flex h-5 w-full items-center justify-between gap-2 rounded-[13px] bg-white px-[9px] py-[2px]">
+          <div className="h-1 grow rounded-md bg-slate-100">
+            <ProgressBar
+              completed={progress}
+              bgColor="#0F172A"
+              isLabelVisible={false}
+              baseBgColor="transparent"
+              transitionDuration="500ms"
+              height="100%"
+            />
+          </div>
+          <div className="text-xs font-semibold leading-4">{progress}%</div>
+        </div>
+        <div className="mt-4 flex flex-col text-sm text-slate-800 tablet:flex-row tablet:gap-4 desktop:gap-6">
+          <TodoSection
+            title="To do"
+            placeholder="아직 해야할 일이 없어요"
+            fetchNextPage={fetchNextTodosPage}
+            hasNextPage={hasTodosNextPage}
+            isFetching={todoIsFetching}
+            isToggleOpen={isToggleOpen}
+            todos={todosInfo}
+            totalCount={todosTotalCount}
+          />
+          <TodoSection
+            title="Done"
+            placeholder="아직 다 한 일이 없어요"
+            fetchNextPage={fetchNextDonesPage}
+            hasNextPage={hasDonesNextPage}
+            isFetching={doneIsFetching}
+            isToggleOpen={isToggleOpen}
+            todos={donesInfo}
+            totalCount={donesTotalCount}
           />
         </div>
-        <div className="text-xs font-semibold leading-4">{progress}%</div>
-      </div>
-      <div className="mt-4 flex flex-col text-sm text-slate-800 tablet:flex-row tablet:gap-4 desktop:gap-6">
-        <TodoSection
-          title="To do"
-          placeholder="아직 해야할 일이 없어요"
-          todos={todos}
-          handleToggleDone={handleToggleDone}
-          handleTodoClick={handleTodoClick}
-        />
-        <TodoSection
-          title="Done"
-          placeholder="아직 다 한 일이 없어요"
-          todos={dones}
-          handleToggleDone={handleToggleDone}
-          handleTodoClick={handleTodoClick}
-        />
       </div>
       {(isTodosMoreThanFive || isDonesMoreThanFive) && (
-        <button
-          type="button"
-          onClick={() => setIsToggleOpen((prev) => !prev)}
-          className="mt-3 flex w-full justify-center"
-        >
-          <div className="flex h-8 w-[120px] items-center rounded-2xl bg-white">
-            {isToggleOpen ? (
-              <>
-                <div className="ml-10 text-sm font-semibold leading-5 text-slate-700">
-                  닫기
-                </div>
-                <ArrowUpIcon />
-              </>
-            ) : (
-              <>
-                <div className="ml-8 text-sm font-semibold leading-5 text-slate-700">
-                  더보기
-                </div>
-                <ArrowDownIcon />
-              </>
-            )}
-          </div>
-        </button>
+        <ToggleButton
+          isToggleOpen={isToggleOpen}
+          setIsToggleOpen={setIsToggleOpen}
```

Comment:
[Suggestion] 이렇게 하는것도 괜찮을거 같습니동
```suggestion
          onToggleOpen={(value)=>setIsToggleOpen(value)}
```